scripts/pysa_fuzzer/run.py

```diff
@@ -0,0 +1,148 @@
+
+def find_undetected_files():
+    # Load the number of files from the temp file
+    try:
+        with open('../num_files.tmp', 'r') as tmp_file:
+            num_files = int(tmp_file.read().strip())
+    except FileNotFoundError:
+        logging.error("Temporary file with number of files not found.")
+        return
+
+    # Load the analysis output from the file
+    try:
+        with open('../analysis_output.tmp', 'r') as file:
+            analysis_output = json.load(file)
+    except FileNotFoundError:
+        logging.error("Analysis output file not found.")
+        return
+
+    # Extract the list of files where the flow has been detected
+    detected_files = set()
+    for entry in analysis_output:
+        detected_files.add(os.path.basename(entry['path']))  # Normalize the path to get the filename only
+
+    # Generate the full list of files (from test_1.py to test_{num_files}.py)
+    all_files = {f"test_{i}.py" for i in range(1, num_files + 1)}
+
+    # Find the files where the flow has not been detected
+    undetected_files = all_files - detected_files
+
+    # Output the list of undetected files
+    logging.info("Files where the flow has not been detected:")
+    for file in sorted(undetected_files, key=lambda x: int(x[len('test_'):-len('.py')])):
+        logging.info(file)
+
+    # Calculate and print the percentage of undetected files
+    total_files = len(all_files)
+    undetected_count = len(undetected_files)
+    undetected_percentage = (undetected_count / total_files) * 100
+    logging.info(f"Flow has not been detected in {undetected_percentage:.2f}% of the files")
+
+def clean_up():
+    files_to_remove = ['sources_sinks.pysa', 'taint.config', '.pyre_configuration', 'analysis_output.tmp', 'num_files.tmp']
+    for file in files_to_remove:
+        try:
+            os.remove(file)
+        except FileNotFoundError:
+            logging.warning(f"{file} not found for cleanup.")
+
+    dirs_to_remove = ['generated_files', '__pycache__']
+    for dir in dirs_to_remove:
+        shutil.rmtree(dir, ignore_errors=True)
+
+def main():
+    parser = argparse.ArgumentParser(description="Build script with setup, analysis, and cleanup.")
+    parser.add_argument('action', choices=['all', 'analyze', 'clean'], help="Action to perform")
+    parser.add_argument('--num-files', type=int, default=100, help="Number of files to generate")
+    parser.add_argument('--num-statements', type=int, default=20, help="Number of statements to generate in each file")
+
+    args = parser.parse_args()
+
+    if args.action == 'all':
+        generate_python_files(args.num_files, args.num_statements)
+        configure_and_analyze()
+        run_pyre()
+        # Save the number of files to a temporary file
+        with open('num_files.tmp', 'w') as tmp_file:
+            tmp_file.write(str(args.num_files))
+    elif args.action == "analyze":
+        find_undetected_files()

```

Comment:
What does action `analyze` mean? If it means finding the files where Pysa can't find a flow, maybe a more descriptive name is just `find-undetected-files`? But this is a bit too long -- naming is hard. Feel free to suggest alternatives -- any more descriptive names would be better.

Reply:
I believe just find-undetected suffices.